honeybee/aperture.py

```diff
@@ -0,0 +1,506 @@
+        """
+        assert isinstance(other_aperture, Aperture), \
+            'Expected Aperture. Got {}.'.format(type(other_aperture))
+        self.boundary_condition = Surface(other_aperture)
+        other_aperture.boundary_condition = Surface(self)
+
+    def overhang(self, depth, angle=0, indoor=False):
+        """Get a single overhang for this Aperture.
+
+        Args:
+            depth: A number for the overhang depth.
+            angle: A number for the for an angle to rotate the overhang in degrees.
+                Default is 0 for no rotation.
+            indoor: Boolean for whether the overhang should be generated facing the
+                opposite direction of the aperture normal (typically meaning
+                indoor geometry). Default: False.
+
+        Returns:
+            overhang: A Shade object for the Aperture overhang. Will be None if the
+                Aperture geometry does not have a flat top.
+        """
+        overhang = self.generate_louvers_by_number(
+            1, depth, angle=angle, indoor=indoor)
+        return overhang[0] if len(overhang) != 0 else None
+
+    def right_fin(self, depth, angle=0, indoor=False):
+        """Get a single vertical fin on the right side of this Aperture.
+
+        Args:
+            depth: A number for the fin depth.
+            angle: A number for the for an angle to rotate the fin in degrees.
+                Default is 0 for no rotation.
+            indoor: Boolean for whether the fin should be generated facing the
+                opposite direction of the aperture normal (typically meaning
+                indoor geometry). Default: False.
+
+        Returns:
+            fin: A Shade object for the Aperture fin. Will be None if the
+                Aperture geometry does not have a flat right side.
+        """
+        fin = self.generate_louvers_by_number(
+            1, depth, angle=angle, contour_vector=Vector3D(1, 0, 0), indoor=indoor)
+        return fin[0] if len(fin) != 0 else None
+
+    def left_fin(self, depth, angle=0, indoor=False):
+        """Get a single vertical fin on the left side of this Aperture.
+
+        Args:
+            depth: A number for the fin depth.
+            angle: A number for the for an angle to rotate the fin in degrees.
+                Default is 0 for no rotation.
+            indoor: Boolean for whether the fin should be generated facing the
+                opposite direction of the aperture normal (typically meaning
+                indoor geometry). Default: False.
+
+        Returns:
+            fin: A Shade object for the Aperture fin. Will be None if the
+                Aperture geometry does not have a flat right side.
+        """
+        fin = self.generate_louvers_by_number(
+            1, depth, angle=angle, contour_vector=Vector3D(1, 0, 0),
+            flip_start_side=True, indoor=indoor)
+        return fin[0] if len(fin) != 0 else None
+
+    def extruded_border(self, depth, indoor=False):
```

Comment:
This will be very useful for creating daylight models from energy models. Specially now that we can get the thickness of the wall construction.